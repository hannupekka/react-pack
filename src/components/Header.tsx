import styles from '@app/styles/components/Header.less';
import * as React from 'react';
import { Link } from 'react-router-dom';
import { pure } from 'recompose';

const logo = require('../assets/logo.png');

const Header = () => (
  <header className={styles.header}>
    <h1 className={styles.title}>Example application</h1>
    <img className={styles.logo} src={logo} alt="" />
    <nav className={styles.navigation}>
      <ul>
        <li>
          <Link to="/">Index</Link>
        </li>
        <li>
          <Link to="/repos">Repo search</Link>
        </li>
      </ul>
    </nav>
  </header>
);

export default pure(Header);
