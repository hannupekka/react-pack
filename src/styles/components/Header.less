@import '~styles/main';

.header {
  background-color: @color__gray--light;
  padding: @padding;
}

.title {
  margin-top: 0;
}

.navigation {
  ul {
    list-style-type: none;
    padding: 0;

    li {
      display: inline-block;
    }

    li + li {
      margin-left: @padding * 2;
    }
  }
}

.logo {
  display: none;
}

@media @md {
  .logo {
    display: block;
  }
}
