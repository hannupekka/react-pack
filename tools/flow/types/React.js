declare type ElementType = React$Element<any>
