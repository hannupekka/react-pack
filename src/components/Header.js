import styles from 'styles/components/Header';
import React from 'react';
import { pure } from 'recompose';
import { Link } from 'react-router';
import CSSModules from 'react-css-modules';

const logo = require('../assets/logo.png');

const Header = () => {
  return (
    <header styleName="header">
      <h1>Example application</h1>
      <img src={logo} role="presentation" />
      <nav>
        <ul>
          <li><Link to={'/'}>Index</Link></li>
          <li><Link to={'/image'}>Image</Link></li>
        </ul>
      </nav>
    </header>
  );
};

export default pure(CSSModules(Header, styles));
