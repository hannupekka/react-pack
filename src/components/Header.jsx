import styles from 'styles/components/Header.less';
import React from 'react';
import { Link } from 'react-router-dom';
import { hot } from 'react-hot-loader';
import CSSModules from 'react-css-modules';

import logo from '../assets/logo.png';

const Header = () => (
  <header styleName="header">
    <h1 styleName="title">Example application</h1>
    <img styleName="logo" src={logo} alt="" />
    <nav styleName="navigation">
      <ul>
        <li>
          <Link to="/">Index</Link>
        </li>
        <li>
          <Link to="/repos">Repo search</Link>
        </li>
      </ul>
    </nav>
  </header>
);

export default hot(module)(CSSModules(Header, styles));
