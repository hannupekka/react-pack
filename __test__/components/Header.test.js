import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Header from 'components/Header';

describe('Header', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <Header />
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
