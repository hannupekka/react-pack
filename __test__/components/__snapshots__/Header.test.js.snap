// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  className="header"
>
  <h1
    className="title"
  >
    Example application
  </h1>
  <img
    alt=""
    className="logo"
    src="test-file-stub"
  />
  <nav
    className="navigation"
  >
    <ul>
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Index
        </a>
      </li>
      <li>
        <a
          href="/repos"
          onClick={[Function]}
        >
          Repo search
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
